core/object/object.cpp

```diff
@@ -1344,14 +1341,18 @@ Error Object::connect(const StringName &p_signal, const Callable &p_callable, ui
 		}
 	}
 
+	Object *target_object = p_callable.get_object();
+
 	SignalData::Slot slot;
 
 	Connection conn;
 	conn.callable = target;
 	conn.signal = ::Signal(this, p_signal);
 	conn.flags = p_flags;
 	slot.conn = conn;
-	slot.cE = target_object->connections.push_back(conn);
+	if (target_object) {
```

Comment:
I was about to ask about how changes to this function will interact with callables of static functions. But I think this indicates that you have them in mind, haven't you?

Reply:
Yeah, the aim of this PR is to fix situations where callables created via `callable_mp_static()` are not accepted (because `get_object()` returns null), but that technically _should_ be able to work.